jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -215,24 +272,25 @@ public void tag(ImageReference originalImageReference, ImageReference newImageRe
   }
 
   /**
-   * Gets the size of an image in the Docker daemon.
+   * Gets the size, image ID, and diff IDs of an image in the Docker daemon.
    *
-   * @param imageReference the image to find the size of
-   * @return the size in bytes
-   * @throws IOException if an I/O exception occurs
-   * @throws InterruptedException if the 'docker inspect' process is interrupted
+   * @param imageReference the image to inspect
+   * @return the size, image ID, and diff IDs of the image
+   * @throws IOException if an I/O exception occurs or {@code docker inspect} failed
+   * @throws InterruptedException if the {@code docker inspect} process was interrupted
+   * @throws DigestException if parsing the image ID or diff IDs failed
    */
-  public long sizeOf(ImageReference imageReference) throws IOException, InterruptedException {
-    Process sizeProcess = docker("inspect", "-f", "{{.Size}}", imageReference.toString());
-
-    if (sizeProcess.waitFor() != 0) {
+  public InspectResults inspect(ImageReference imageReference)
+      throws IOException, InterruptedException, DigestException {
+    Process inspectProcess =
+        docker("inspect", "-f", "{{.Size}},{{.Id}},{{.RootFS.Layers}}", imageReference.toString());
```

Comment:
You could use `{{range .RootFS.Layers}}{{.}},{{end}}` in the format to simplify the parsing and treat items 3..N as the layer diffsIds?

Is it worth adding [`--type image`](https://github.com/docker/cli/blob/master/docs/reference/commandline/inspect.md#specify-target-type---type) to avoid any possibility of the user providing a name to something that is not an image?

Reply:
Looking at this again, I'm thinking it might be simpler to just format the docker output as a JSON string and parse that instead. e.g. `docker inspect -f '{"size":{{.Size}},"imageId":"{{.Id}}","diffIds":{{json .RootFS.Layers}}}'`. That way `parseInspectResults` just becomes `return JsonTemplateMapper.readJson(output, InspectResults.class);`

The method already takes an `ImageReference`, so I think it'd be strange for the user to pass in the name of a non-image here. Not sure the extra validation is necessary.

Reply:
I don't think the use of `--type image` hurts.  But the possibility of clashes does seem small: volumes and containers can't have `/` in their names, and the people are unlikely to use official docker images like `openjdk` without a tag (I hope).

Reply:
Without `--type image` we get a confusing error if we have a non-image object by the same name:
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.6.2-SNAPSHOT:dockerBuild (default-cli) on project first: 'docker inspect' command failed with error: Template parsing error: template: :1:10: executing "" at <.Size>: map has no entry for key "Size"
```
With --type image` we get a more obvious error:
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.6.2-SNAPSHOT:dockerBuild (default-cli) on project first: 'docker inspect' command failed with error: Error: No such image: openjdk
```

`--type image` has an added bonus as it uses `No such *image*` in the error.  If we use non-existent object:
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.6.2-SNAPSHOT:dockerBuild (default-cli) on project first: 'docker inspect' command failed with error: Error: No such object: non-existant
```

I think the more meaningful error warrants the extra two arguments.